_studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_ddi.h

```diff
@@ -194,13 +194,23 @@ typedef struct tagENCODE_CAPS_VP9
         mfxU32 bufferSize)
     {
         mfxU32   ivf_file_header[8] = { 0x46494B44, 0x00200000, 0x30395056, width + (height << 16), FrameRateN, FrameRateD, numFramesInFile, 0x00000000 };
-        memcpy_s(pBitstream, bufferSize, ivf_file_header, sizeof (ivf_file_header));
+        if (bufferSize < sizeof(ivf_file_header))
+            return MFX_ERR_MORE_DATA;
+
+        std::copy(std::begin(ivf_file_header),std::end(ivf_file_header), pBitstream);
+
+        return MFX_ERR_NONE;
     };
 
-    inline void AddPictureHeader(mfxU8* pBitstream, mfxU32 bufferSize)
+    inline mfxStatus AddPictureHeader(mfxU8* pBitstream, mfxU32 bufferSize)
     {
         mfxU32 ivf_frame_header[3] = { 0x00000000, 0x00000000, 0x00000000 };
-        memcpy_s(pBitstream, bufferSize, ivf_frame_header, sizeof (ivf_frame_header));
+        if (bufferSize < sizeof(ivf_frame_header))
+            return MFX_ERR_MORE_DATA;
+
+        std::fill_n(pBitstream, sizeof(ivf_frame_header), 0);
```

Comment:
I would still prefer std::copy for unified style, but this one is equivalent now

Reply:
This is Leonids idea. Looks like nice:)

Reply:
Then what for is array initialization, if data is never used. Possible to catch a compiler warning.

Reply:
the array is totally useless now...
if you really want to use std::fill_n, do something like this:

```
inline mfxStatus AddPictureHeader(mfxU8* pBitstream, mfxU32 bufferSize) 
{
    mfxU32 number_of_zero_bytes = 3*sizeof(mfxU32);

    if (bufferSize < number_of_zero_bytes)
            return MFX_ERR_MORE_DATA;

     std::fill_n(pBitstream, number_of_zero_bytes, 0);

     return MFX_ERR_NONE;
}
```